server/bot/bot.go

```diff
@@ -0,0 +1,132 @@
+package bot
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	n "github.com/dyng/nosdaily/nostr"
+	"github.com/dyng/nosdaily/types"
+	"github.com/nbd-wtf/go-nostr"
+	"github.com/nbd-wtf/go-nostr/nip19"
+)
+
+var userSubStore map[string]string
+
+func init() {
+	userSubStore = make(map[string]string)
+}
+
+type BotApplication struct {
+	bot    *Bot
+	config *types.Config
+}
+
+type Bot struct {
+	client *n.Client
+	sk     string
+	pub    string
+}
+
+func NewBotApplication(config *types.Config) *BotApplication {
+	ctx := context.Background()
+
+	client, err := n.NewClient(ctx, config.Bot.Relays)
+	if err != nil {
+		panic(err)
+	}
+
+	bot, err := NewBot(ctx, client, config.Bot.SK)
+	if err != nil {
+		panic(err)
+	}
+
+	return &BotApplication{
+		bot:    bot,
+		config: config,
+	}
+}
+
+func (ba *BotApplication) Run(ctx context.Context) error {
+	c, err := ba.bot.Listen(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println("[bot] Start listening to subscribe messages...")
```

Comment:
Can be replaced by `github.com/ethereum/go-ethereum/log`. We use this logging library for prototyping and can switch to others later.